### PR TITLE
attitude setpoint topic cleanup:

### PR DESCRIPTION
**Pull request by tumbili**
- removed redundant and unused data out of attitude setpoint topic
- convert attitude representations in the code base

Signed-off-by: Roman <bapstroman@gmail.com>

Replaces #4478 

Tested all modes in SITL for FW and MC.

### CODE REVIEW

File: src/modules/sdlog2/sdlog2.c [hunk 1]
```diff
@@ -2244,19 +2244,23 @@ int sdlog2_thread_main(int argc, char *argv[])
...
+			log_msg.body.log_ATT.yaw = atan2f(2*(q0*q3 + q1*q2), 1 - 2*(q2*q2 + q3*q3));
 			log_msg.body.log_ATT.roll_rate = buf.att.rollspeed;
 			log_msg.body.log_ATT.pitch_rate = buf.att.pitchspeed;
 			log_msg.body.log_ATT.yaw_rate = buf.att.yawspeed;
-			log_msg.body.log_ATT.gx = buf.att.g_comp[0];
-			log_msg.body.log_ATT.gy = buf.att.g_comp[1];
-			log_msg.body.log_ATT.gz = buf.att.g_comp[2];
+			log_msg.body.log_ATT.gx = 0;
+			log_msg.body.log_ATT.gy = 0;
+			log_msg.body.log_ATT.gz = 0;
```
Comment (julianoes) [inline src/modules/sdlog2/sdlog2.c]: You don't want to delete these values from the log since they are now obsolete?

**Comment by julianoes** [comment]
Reviewed, didn't see any silly mistakes.

File: src/modules/sdlog2/sdlog2.c (hunk 1 above)
Comment (tumbili) [inline src/modules/sdlog2/sdlog2.c]: @julianoes Yes, will do!

Comment (julianoes) [inline src/modules/sdlog2/sdlog2.c]: Unless we break all log scripts, not sure...

Comment (tumbili) [inline src/modules/sdlog2/sdlog2.c]: @julianoes I don't think it's going to be an issue. And if yes, then we fix it :-)

**Comment by julianoes** [comment]
@tumbili Can you fix the thing you promised please, and give it a small bench test so we can get it in?

**Comment by LorenzMeier** [comment]
As far as I can see this PR missed the change in msg/vehicle_attitude. Can we get that in, complete it and test-flight on real vehicles as well? Having the output of "top" in a bench test of master and this rebased on master would be important so we can rule out CPU load side-effects.

**Comment by LorenzMeier** [comment]
@tumbili Can you rebase this so we can get it in?

**Comment by LorenzMeier** [comment]
replaced by https://github.com/PX4/Firmware/pull/4312

**Comment by julianoes** [comment]
Delete branch?

**Pull request closed without merging**